MatchGame/Game.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MatchGame
+{
+    public class Game : IGame
+    {
+        private readonly IDeck _gameDeck;
+        private readonly IPlayer _player1;
+        private readonly IPlayer _player2;
+        private MatchRuleEnum _matchRule;
+        private readonly Stack<Card> _pile = new();
+
+        public Game(IDeck gameDeck, IPlayer player1, IPlayer player2)
```

Comment:
Passing the players through the constructor felt a bit off. Maybe better passing through method injection.